apps/oc_chef_wm/src/oc_chef_wm_authn_ldap.erl

```diff
@@ -142,21 +142,27 @@ maybe_encrypt_session(start_tls, {ok, Session}, Timeout) ->
 maybe_encrypt_session(_, {ok, Session}, _) ->
     {ok, Session}.
 
-
+-spec canonical_username(string()) -> binary().
+canonical_username(Username) ->
+    characters_to_binary(
+      re:replace(string:to_lower(Username),
+                 "[^0-z0-9_-]",
+                 "_",
+                 [{return, list}, global])).
 
 result_to_user_ejson(_, UserName, []) ->
     lager:info("User ~p not found in LDAP", [UserName]),
     {error, unauthorized};
-result_to_user_ejson(LoginAttr, _, [{eldap_entry, CN, DataIn}|_]) ->
+result_to_user_ejson(LoginAttr, UserName, [{eldap_entry, CN, DataIn}|_]) ->
 
     % No guarantees on casing, so let's not make assumptions:
     Data = [ { string:to_lower(Key), Value} || {Key, Value} <- DataIn ],
 
-    % loginattr was used to find this record, so we know it must exist
-    [UserName0] = proplists:get_value(LoginAttr, Data),
-    UserName1 = string:to_lower(UserName0),
-    UserName2 = re:replace(UserName1, "[^0-z0-9_-]", "_", [{return, list}, global]),
-    UserName = characters_to_binary(UserName2),
+    % loginattr was used to find this record, so we know it must exist;
+    % however, multiple LoginAttr fields may exist in the LDAP record, take
+    % the first that matches the given UserName
+    [CanonicalUserName|_] = [ canonical_username(U) || U <- proplists:get_value(LoginAttr, Data),
```

Comment:
We took a look at the original code - it actually just grabbed the first name returned which *can* be different than the one provided to login.  So we'll want to continue to ignore the UserName argument to the function, and instead: 

    CanonicalUserName = canonical_username(value_of(LoginAttr, Data))